src/main/scala/higherkindness/skeuomorph/protobuf/Native.scala

```diff
@@ -0,0 +1,349 @@
+
+      NativeOneOfField(name = oneof.getName, tpe = NativeOneOf(name = oneof.getName, fields = oneOfFields))
+    }
+  }
+
+  def fromFieldType(field: FieldDescriptorProto, files: List[FileDescriptorProto]): NativeDescriptor =
+    field.getType match {
+      case Type.TYPE_BOOL     => NativeBool()
+      case Type.TYPE_BYTES    => NativeBytes()
+      case Type.TYPE_DOUBLE   => NativeDouble()
+      case Type.TYPE_FIXED32  => NativeFixed32()
+      case Type.TYPE_FIXED64  => NativeFixed64()
+      case Type.TYPE_FLOAT    => NativeFloat()
+      case Type.TYPE_INT32    => NativeInt32()
+      case Type.TYPE_INT64    => NativeInt64()
+      case Type.TYPE_SFIXED32 => NativeFixed32()
+      case Type.TYPE_SFIXED64 => NativeFixed64()
+      case Type.TYPE_SINT32   => NativeInt32()
+      case Type.TYPE_SINT64   => NativeInt64()
+      case Type.TYPE_STRING   => NativeString()
+      case Type.TYPE_UINT32   => NativeInt32()
+      case Type.TYPE_UINT64   => NativeInt64()
+      case Type.TYPE_ENUM =>
+        findEnum(field.getTypeName, files)
+          .fold[NativeDescriptor](NativeNull())(e => NativeNamedType(e.getName))
+      case Type.TYPE_MESSAGE =>
+        findMessage(field.getTypeName, files)
+          .fold[NativeDescriptor](NativeNull())(e => NativeNamedType(e.getName))
+      case _ => NativeNull()
+    }
+
+  def fromFieldOptionsMsg(options: FieldOptions): List[NativeOption] =
+    NativeOption("deprecated", options.getDeprecated.toString) ::
+      options.getUninterpretedOptionList.j2s.map(t => NativeOption(toString(t.getNameList.j2s), t.getIdentifierValue))
+
+  def fromFieldOptionsEnum(options: EnumOptions): List[NativeOption] = {
+    List(
+      NativeOption("allow_alias", options.getAllowAlias.toString),
+      NativeOption("deprecated", options.getDeprecated.toString)) ++
+      options.getUninterpretedOptionList.j2s.map(t => NativeOption(toString(t.getNameList.j2s), t.getIdentifierValue))
+  }
+
+  def toString(nameParts: Seq[NamePart]): String =
+    nameParts.foldLeft("")((l, r) => if (r.getIsExtension) s"$l.($r)" else s"$l.$r")
+
+  def findMessage(name: String, files: List[FileDescriptorProto]): Option[DescriptorProto] = {
+    case class NamedMessage(fullName: String, msg: DescriptorProto)
+    val all: List[NamedMessage] = files.flatMap(f =>
+      f.getMessageTypeList.j2s.flatMap(m =>
+        NamedMessage(s".${f.getPackage}.${m.getName}", m) :: m.getNestedTypeList.j2s.map(n =>
+          NamedMessage(s".${f.getPackage}.${m.getName}.${n.getName}", n))))
+    all.find(_.fullName == name).map(_.msg)
+  }
+
+  def findEnum(name: String, files: List[FileDescriptorProto]): Option[EnumDescriptorProto] = {
+    case class NamedEnum(fullName: String, msg: EnumDescriptorProto)
+    files
+      .flatMap(f => f.getEnumTypeList.j2s.map(m => NamedEnum(s".${f.getPackage}.${m.getName}", m)))
+      .find(_.fullName == name)
+      .map(_.msg)
+  }
+
+  implicit class LabelOps(self: Label) {
```

Comment:
Can we make this implicit class private? I'd like to avoid polluting the namespace in case someone imports `NativeField._`.
```suggestion
  private implicit class LabelOps(self: Label) {
```

Reply:
Solved at https://github.com/higherkindness/skeuomorph/pull/66/commits/78581e098293b077fa76b8987819e49fba0d4703